app/src/main/java/com/rockthevote/grommet/ui/registration/BaseRegistrationFragment.java

```diff
@@ -30,13 +31,16 @@
     @Inject
     RegistrationDao registrationDao;
 
+    @Inject
+    SessionDao sessionDao;
+
     @Override
     public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
         Injector.obtain(getActivity()).inject(this);
         viewModel = new ViewModelProvider(
                 requireActivity(),
-                new RegistrationViewModelFactory(registrationDao)
+                new RegistrationViewModelFactory(registrationDao, sessionDao)
```

Comment:
I was hoping you'd only need to pass in the `sessionDao` and then be able to use the [entity relationship](https://developer.android.com/training/data-storage/room/relationships#one-to-many) `SessionWithRegistrations` to get all registrations for that session via the `getSessionWithRegistrations()` method (might have to modify the select query).

Conversely, you could probably make an inverse relationship and be able to just use the `registrationDao`. 

Again though, given the short SLA for this version, not super important.

Reply:
It would definitely be better to do it that way, but where this is at, I don't have sessions object yet, so can't construct a `SessionWithRegistrations`

Reply:
Looking at this more, this is for the construction of the RegistrationViewModel, is it possible to add a registration to the Registration table without the RegistrationDao? I think we may need both for this VM